afkak/consumer.py

```diff
@@ -720,8 +744,7 @@ def _handle_fetch_error(self, failure):
             log.debug(
                 "%r: Exhausted attempts: %d fetching messages from kafka: %r",
                 self, self.request_retry_max_attempts, failure)
-            self._start_d.errback(failure)
-            return
+
```

Comment:
This change (above) is why that test is failing.  The consumer has decided that it's run out of attempts, but you removed the errback() call and the early return.

Reply:
Thanks @rthille!